AIMSICD/src/main/java/com/secupwn/aimsicd/ui/activities/MainActivity.java

```diff
@@ -281,6 +290,27 @@ void selectDrawerItem(int position) {
         }
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == ACTIVITY_RESULT_SELECT_CELLTOWERS) {
+            if (resultCode == RESULT_OK) {
+                log.debug("Chosen file: " + data.getDataString());
+                try {
+                    String type = getContentResolver().getType(data.getData());
+                    boolean isGzip = type != null && type.equals("application/octet-stream") &&
+                            data.getDataString().endsWith(".gz");
+                    InputStream importFile = getContentResolver().openInputStream(data.getData());
```

Comment:
Im not sure if this is the right place to open the `InputStream` and to check the type.

I'd prefer to pass `data.getData()` around instead of the open `InputStream`